### PR TITLE
fix: Fix bug in RemoteLauncher with SageMakerBackend

### LINKED ISSUE #716: examples/launch_height_sagemaker_remotely.py does not work if distribute_trials_on_sagemaker = True
**Describe the bug**
The script uses `RemoteLauncher` to launch an experiment remotely, one can select between local and SageMaker backend. The default is local, and the script works.

But if the SageMaker backend is chosen instead, the script fails.

What happens is that while `tuner.trial_backend.tuner_name` is set properly on the host the script is executed, it becomes `None` inside the SageMaker training job. The backend then tries to launch a trial training job of name "0-None", which fails.

`self.tuner_name` is initialized to `None` when `SageMakerBackend` is created, but it set when the `Tuner` object is created. The `RemoteLauncher` serializes the `Tuner` object along with backend and all other dependencies. Just before serialization, the `self.tuner_name` is properly set. But when this object is deserialized on the client side, `self.tuner_name` of the backend is `None`.

**To reproduce**  
2. Run the following command '....'
3. See error

**Expected behavior**  

**Paste the output of the `pip freeze` command below:**

**Desktop (please complete the following information):**  
 - OS: [e.g. MacOS Monterey 12.6.1 / Ubuntu Linux / Windows 11]
 - Syne Tune Version [e.g. v0.3.4]

**Screenshots**  

**Additional context**

### PR DESCRIPTION
**Pull request by mseeger**
Fixes a bug (and adds test) which blocks one of our customers.

Apart from this, I also replaced `logging` with `logger` at several places.

Closes #716 

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: syne_tune/backend/sagemaker_backend/sagemaker_backend.py [hunk 1]
```diff
@@ -522,8 +529,12 @@ def delete_checkpoint(self, trial_id: int):
     def set_path(
         self, results_root: Optional[str] = None, tuner_name: Optional[str] = None
     ):
-        # we use the tuner-name to set the checkpoint directory
-        self.tuner_name = tuner_name
+        """
+        For this backend, it is mandatory to call this method passing ``tuner_name``
+        before the backend is used. ``results_root`` is ignored here.
+        """
+        if tuner_name is not None:
```
Comment (mseeger) [inline syne_tune/backend/sagemaker_backend/sagemaker_backend.py:536]: This is the bugfix. This is called in `remote_main.py`, and it set `self.tuner_name` back to `None`, which is clearly wrong.

File: examples/launch_height_sagemaker_remotely.py [hunk 2]
```diff
@@ -42,6 +43,11 @@
 if __name__ == "__main__":
     logging.getLogger().setLevel(logging.INFO)
 
+    parser = ArgumentParser()
```
Comment (mseeger) [inline examples/launch_height_sagemaker_remotely.py:46]: Allows to run this example w/o SM backend, so it is tested.

**Review by jgolebiowski — APPROVED** [review]
LGTM.

**Comment by mseeger** [comment]
https://github.com/awslabs/syne-tune/actions/runs/5277103643

end-to-end workflow runs

**Pull request merged**